src/controllers/grantAWSAccessCommand.ts

```diff
@@ -0,0 +1,39 @@
+import { discordTextResponse } from "../utils/discordResponse";
+import { SUPER_USER_ONE, SUPER_USER_TWO } from "../constants/variables";
+import { env } from "../typeDefinitions/default.types";
+import {
+  messageRequestMember,
+  messageRequestDataOptions,
+} from "../typeDefinitions/discordMessage.types";
+import { grantAWSAccess } from "../utils/awsAccess";
+import { DevFlag } from "../typeDefinitions/filterUsersByRole";
+
+export async function grantAWSAccessCommand(
+  transformedArgument: {
+    member: messageRequestMember;
+    userDetails: messageRequestDataOptions;
+    awsGroupDetails: messageRequestDataOptions;
+    channelId: number;
+    dev?: DevFlag;
+  },
+  env: env,
+  ctx: ExecutionContext
+) {
+  const dev = transformedArgument?.dev?.value || false;
+  if (dev) {
+    const isUserSuperUser = [SUPER_USER_ONE, SUPER_USER_TWO].includes(
+      transformedArgument.member.user.id.toString()
+    );
+    if (!isUserSuperUser) {
+      const responseText = `You're not authorized to make this request.`;
+      return discordTextResponse(responseText);
+    }
+    const roleId = transformedArgument.userDetails.value;
+    const groupId = transformedArgument.awsGroupDetails.value;
+    const channelId = transformedArgument.channelId;
+
+    return grantAWSAccess(roleId, groupId, env, ctx, channelId);
+  } else {
+    return discordTextResponse("Please enable feature flag to make this work");
+  }
```

Comment:
Can we do an early return when `dev` is false and then do the operations we want to actually do? (for better readability)
Something like this
```
if(!dev) {
    return discordTextResponse("Please enable feature flag to make this work");
}
// Do the actual operations (which is all the lines in the else block)
```

Reply:
updated